reducer/union.go

```diff
@@ -0,0 +1,91 @@
+package reducer
+
+import (
+	"github.com/brimsec/zq/expr"
+	"github.com/brimsec/zq/zcode"
+	"github.com/brimsec/zq/zng"
+	"github.com/brimsec/zq/zng/resolver"
+)
+
+type Union struct {
+	Reducer
+	zctx *resolver.Context
+	arg  expr.Evaluator
+	typ  zng.Type
+	val  map[string]struct{}
+	size int
+}
+
+func newUnion(zctx *resolver.Context, arg, where expr.Evaluator) *Union {
+	return &Union{
+		Reducer: Reducer{where: where},
+		zctx:    zctx,
+		arg:     arg,
+		val:     make(map[string]struct{}),
+	}
+}
+
+func (u *Union) Consume(r *zng.Record) {
+	if u.filter(r) {
+		return
+	}
+	v, err := u.arg.Eval(r)
+	if err != nil || v.IsNil() {
+		return
+	}
+	if u.typ == nil {
+		u.typ = v.Type
+	} else if u.typ != v.Type {
+		u.TypeMismatch++
+		return
+	}
+	u.update(v.Bytes)
+}
+
+func (u *Union) update(b zcode.Bytes) {
+	if _, ok := u.val[string(b)]; !ok {
+		u.val[string(b)] = struct{}{}
+		u.size += len(b)
+		for u.size > MaxValueSize {
+			u.deleteOne()
+			u.MemExceeded++
+		}
+	}
+}
+
+func (u *Union) deleteOne() {
+	for key := range u.val {
+		u.size -= len([]byte(key))
```

Comment:
```suggestion
		u.size -= len(key)
```

Reply:
Ok, I was thinking go did rune length on strings but I see it doesn't.  Thanks!